_posts/2023-09-13-prospero-add-feature-pack.adoc

```diff
@@ -0,0 +1,269 @@
+
+[source]
+----
+Installing feature pack `org.wildfly:wildfly-datasources-galleon-pack` in server `/tmp/wildfly-29.0.1.Final`
+Continue adding the feature pack? [y/N]:y
+Applying changes
+
+Feature-packs resolved.
+Packages installed.
+Downloaded artifacts.
+JBoss modules installed.
+Configurations generated.
+JBoss examples installed.
+Operation completed in 19.62 seconds.
+----
+
+After running this command, a new module `org.mariadb.jdbc:main` will be created in the server modules directory and the `standalone/configuration/standalone.xml` 
+will be updated with a new driver and datasource configuration.
+
+== Updating mariadb driver version
+
+Steps above installed a default version of MariaDB driver, but using prospero and wildfly channels we are able to udate that.
+
+First, let's add desired version of the driver to the datasources manifest:
+
+[,bash]
+----
+echo "
+  - groupId: org.mariadb.jdbc
+    artifactId: mariadb-java-client
+    version: 3.2.0
+" >> /tmp/datasources-manifest.yaml
+----
+
+Next, we just need to let prospero perform an update:
+
+[,bash]
+----
+./bin/prospero.sh update perform --dir /tmp/wildfly-29.0.1.Final
+----
+
+The command will report available updates - in this case showing the changed mariadb driver - and ask you to confirm the changes.
+
+[source]
+----
+Updates found:
+  org.mariadb.jdbc:mariadb-java-client                  3.1.4                ==>  3.2.0
+Continue with update [y/N]: y
+Building updates
+
+Feature-packs resolved.
+Packages installed.
+Downloaded artifacts.
+JBoss modules installed.
+Configurations generated.
+JBoss examples installed.
+
+Build update complete!
+Applying updates
+Update complete!
+Operation completed in 25.48 seconds.
+----
+
+Now we can check that the `org.mariadb.jdbc:main` module:
+
+[,bash]
+----
+ls /tmp/wildfly-29.0.1.Final/modules/org/mariadb/jdbc/main
+----
+
+and see the updated driver:
+
+[source]
+----
+mariadb-java-client-3.2.0.jar  module.xml
+----
+
+== Starting the server
+
+Before starting the server with the new datasource, we need to provide the MariaDB connection information. Those can either be 
+added to the server configuration as WildFly system properties, or as OS environment variables. For the purpose of this article, we will assume that 
+the MariaDB database called `testdb` is available on `localhost:3306` and is secured using `test/password` credentials.
+
+The detailed description of available configuration options can be found in the datasource feature pack documentation, 
+eg. https://github.com/wildfly-extras/wildfly-datasources-galleon-pack/blob/main/doc/mariadb/README.md#required-configuration[MariaDB configuration].
+Managing WildFly system properties is described in the https://docs.wildfly.org/16/Admin_Guide.html#properties-2[WildFly Admin Guide]
+
+Let's try to set the required configuration using environment variables and start the server:
+
+[source]
+----
+export MARIADB_USER=test
```

Comment:
it might be worth noting that we did nothing about the host as `localhost:3306` is the default